blockchain/scriptval.go

```diff
@@ -227,7 +244,19 @@ func checkBlockScripts(block *dcrutil.Block, utxoView *UtxoViewpoint, txTree boo
 	}
 	txValItems := make([]*txValidateItem, 0, numInputs)
 	for _, tx := range txs {
-		for txInIdx, txIn := range tx.MsgTx().TxIn {
+		// Skip version 2 revocations if the automatic ticket revocations agenda is
+		// active.  This is allowed since consensus rules enforce that revocations
+		// MUST pay to the addresses specified by the original commitments in the
+		// ticket.
+		msgTx := tx.MsgTx()
+		if isAutoRevocationsEnabled &&
+			stake.IsSSRtx(msgTx, isAutoRevocationsEnabled) &&
+			msgTx.Version == stake.TxVersionAutoRevocations {
```

Comment:
Updated to put the version check first.  It felt more natural to check the type before the version since the version is specific to the type, but agreed that it's a quicker check and can short circuit early in some cases.